### PR TITLE
Reland: Migrate examples new embedding

### PR DESCRIPTION
**Pull request by blasten**
Relands https://github.com/flutter/flutter/pull/43221

### CODE REVIEW

File: dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt [hunk 1]
```diff
@@ -41,7 +44,7 @@ class MainActivity : FlutterActivity() {
 
         // Triggers the Android keyboard, which causes the resize of the Flutter view.
         // We need to wait for the app to complete.
-        MethodChannel(flutterView, "com.example.abstract_method_smoke_test")
+        MethodChannel(flutterEngine.getDartExecutor(), "com.example.abstract_method_smoke_test")
```
Comment (matthew-carroll) [inline dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt]: There has been an API change to `flutterEngine.getDartExecutor().getBinaryMessenger()`. Should we make that change now? Or should we come back later and update?

**Review by matthew-carroll — APPROVED** [review]

File: dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt (hunk 1 above)
Comment (blasten) [reply dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt]: I'd prefer to check this in and then change it again.

Comment (blasten) [reply dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt]: will follow up with that while the device lab tests are running.

**Comment by chinmaygarde** [comment]
Heads up, this commit was flagged as a potential point of regression tracked in https://github.com/flutter/flutter/issues/43688

**Pull request merged**